flexmeasures/data/schemas/sensors.py

```diff
@@ -239,36 +273,58 @@ def _deserialize(
             sensor = SensorIdField(unit=self.to_unit)._deserialize(
                 value["sensor"], None, None
             )
-
             return sensor
 
+        elif isinstance(value, list):
+            if self.return_magnitude is True:
```

Comment:
Actually, this warning is more of developer note that could be logged once upon schema initialization (rather than with every call to deserialize). The behaviour that it warns about is not influenced by whether the user is passing numeric values or string quantities. I'm moving it to `__init__`.

Reply:
Aaaaaand.. moving it back, because the logger cannot be used outside of the app context.